### PR TITLE
Update manu dave

### PR DESCRIPTION
**Pull request by dallenmidd**
Hey Bert,
here are my edits! This looks so good. The writing is really clear, and I think the steps are laid out so clearly. My one small concern is that MEE may want more about why the method we implement is important. Though I am not 100% sure. Maybe it is fine that the Plos One paper does the justifcation of why the method is important, and then this paper is describing the package and showing how it is used. Maybe that is all they are after. But if the intro and conclusion, could just give a little bit of "selling" the method, I think that would be important. Either way it looks great!

I left a couple of notes/comments in there. I have them directly in the text. You can see them as DAVE NOTE:

### CODE REVIEW

File: paper/forestecology.Rmd [hunk 1]
```diff
@@ -213,10 +209,12 @@ growth_scbi <- growth_scbi %>%
...
+The second element of spatial information are blocks corresponding to folds of a spatial cross-validation algorithm used to estimate out-of-sample model error. Conventional cross-validation algorithms assign observations to folds by randomly resampling individual observations. However, an assumption that the observations are independent underlies any cross-validation algorithm. In the case of forest census data, observations exhibit spatial autocorrelation. We therefore incorporate this spatial dependence into the cross-validation algorithm by randomly resampling spatial blocks of trees @roberts_cross-validation_2017 @pohjankukka_estimating_2017. (DAVE NOTE: I don't want to change somethign I don't understand, but "randomly resampling" seems wrong to me. I don't see anything random about our block structure. Seems to me like "incorpoate this spatial dependence into the cross-validation algorithm with our spatial blocks of trees.")
```
Comment (rudeboybert) [inline paper/forestecology.Rmd:212]: you're correct. in regular CV, the observations get assigned randomly to folds. Here, they are assigned depending on their location

File: paper/forestecology.Rmd [hunk 2]
```diff
@@ -213,10 +209,12 @@ growth_scbi <- growth_scbi %>%
...
 
 In the example below, we first manually define four folds that partition the study region as an `sf` object. We then use the output of the `spatialBlock()` function from the `blockCV` package to associate each tree in `growth_scbi` to the correct fold (saved in the `foldID` variable) @valavi_blockcv_2019. ^[In the Appendix we present an example where the folds themselves are also created using the `spatialBlock()` function given a specified `cv_block_size`.] 
 
+DAVE NOTE::there are a couple of places where you convert to factor. Earlier with `sp` and then twice in the code block below. ANy reason we can't just do this "under the hood" in the package code? Also any reason with swtiched from undercase with underscores to `SpatialBlock_scbi` in the naming?
```
Comment (rudeboybert) [inline paper/forestecology.Rmd:216]: huh, that's true. It will make things a lot cleaner. I'll add that to my TODO list of minor pkg code tweaks.

File: paper/forestecology.Rmd [hunk 3]
```diff
@@ -344,12 +342,13 @@ focal_vs_comp_scbi <- focal_vs_comp_scbi %>%
 focal_vs_comp_scbi
 ```
 
-We then compare the observed and fitted/predicted growths to compute the root mean squared error (RMSE) of our model fit. 
+We then compare the observed and fitted/predicted growths to compute the root mean squared error (RMSE) of our model fit. DAVE NOTE: I rounded all the RMSEs, in ecology convention to not show WAY more digits than resolution of the measurement.
```
Comment (rudeboybert) [inline paper/forestecology.Rmd:345]: Agreed on the pseudo-precision. I'm going to see if I can set this globally for the whole document, instead of setting them individually

File: paper/forestecology.Rmd [hunk 4]
```diff
@@ -383,20 +381,21 @@ focal_vs_comp_scbi <- focal_vs_comp_scbi %>%
...
...is end, we use the `foldID` variable defined in Section \ref{spatial-information} whereby all focal trees are assigned to one of `r n_fold` spatially contiguous blocks that act as folds in our cross-validation routine. Figure \ref{fig:scbi-spatial-cross-validation-schematic} presents a schematic illustrating this scheme for fold 1 (bottom-left) as the test set and folds 2, 3, and 4 as the training sets. We fit the model to all focal trees in the training set, apply the model to all focal trees in the test set to compute fitted/predicted values, and compute the RMSE of the observed versus predicted growths. We repeat this procedure 3 more times with each of the three remaining folds acting as the test set and then average all four resulting RMSE's. Furthermore, in order to maintain spatial independence between the test and training set, a fold buffer that extend outwards from the boundary of the test set is computed; all trees falling within this fold buffer are excluded from the training set. DAVE NOTE: I repalced "buffer" with "fold buffer" just to keep clear that this is different than our global buffer. But if you think that distinction is not necessary feel free to switch back!
```
Comment (rudeboybert) [inline paper/forestecology.Rmd:396]: great addition. it disambiguates the inwards vs outwards buffers for study region vs folds nicely

File: paper/forestecology.Rmd [hunk 5]
```diff
@@ -178,9 +174,9 @@ growth_scbi <-
...
 
-The data format of other sites may be such that our `compute_growth()` function doesn't work at all. However, in the end all that matters is that the growth of all trees is saved in a data frame of class `sf` whereby the geolocation of each tree is presented in a `geometry` variable of type `<POINT>` and at a minimum the data contains the variables above.
+The data format of other sites may be such that our `compute_growth()` function doesn't work at all. However, in the end all that matters is that the growth of all stems is saved in a data frame of class `sf` whereby the geolocation of each tree is presented in a `geometry` variable of type `<POINT>` and at a minimum the data contains the variables above (DAVE NOTE: THis isn't exaclty clear to me what variables above means. Maybe list all the variables needed. Do they have to have exact names?)
```
Comment (rudeboybert) [inline paper/forestecology.Rmd:179]: These are names inherent to the sf-point object type; I'll add a note

File: paper/forestecology.Rmd (hunk 2 above)
Comment (rudeboybert) [reply paper/forestecology.Rmd:216]: As for `SpatialBlock_scbi`, my original thought was to keep it consistent with the name of `blockCV::SpatialBlock()`, but yeah I agree, the jarring switch to CamelCase doesn't warrant that.

File: paper/forestecology.Rmd [hunk 6]
```diff
@@ -367,8 +366,7 @@ plot3 <- autoplot(comp_bayes_lm_scbi, type = "competition", sp_to_plot = sp_to_p
 (plot1 | plot2) / plot3
 ```
 
-TODO: Discuss meaning of results.
-
+These plots give the posterior distributions of parameters from equation 1 (DAVE NOTE:SOrry couldn't figure out how to make equation references work). For many pacakge users they will be of interest because they give insight into the species-specific competitive interactions. Setting `type = "intercepts"` gives posterior distributions for $\beta_{0,j}$ and `type = "dbh_slopes"` for $\beta_{dbh,j}$. These give species specific growth independent of competition. The values of more interest are plotted with `type = 'competition'` which gives the posterior distribution for $\lambda_{j,k}$ species-specific competition coefficients (i.e., the $\lambda$-matrix). Negative values indicate a competitor species which slows the growth of a focal species. Here, for example, we see that tulip trees (`litu`) have a strong negative effect on the growth of conspecifics but relatively little effect on neighbors of the other two species.
```
Comment (rudeboybert) [inline paper/forestecology.Rmd:369]: Equation ref fixed

Comment (rudeboybert) [reply paper/forestecology.Rmd:369]: it took me a surprising amount of time to figure out too. here is what I used: https://gist.github.com/dewittpe/ee42f46b23b9de6b1338

**Review by rudeboybert — APPROVED** [review]
Thanks for your comments Dave, they were all excellent. I'll work them locally and then merge them shortly
